crates/ruff_linter/src/rules/refurb/rules/fromisoformat_replace_z.rs

```diff
@@ -0,0 +1,277 @@
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::parenthesize::parenthesized_range;
+use ruff_python_ast::{
+    Expr, ExprAttribute, ExprBinOp, ExprCall, ExprStringLiteral, ExprSubscript, ExprUnaryOp,
+    Number, Operator, UnaryOp,
+};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+use crate::settings::types::PythonVersion;
+
+/// ## What it does
+/// Checks for `datetime.fromisoformat()` calls
+/// where the only argument is an inline replacement
+/// of `Z` with a zero offset timezone.
+///
+/// ## Why is this bad?
+/// On Python 3.11 and later, `datetime.fromisoformat()` can handle most [ISO 8601][iso-8601]
+/// formats (barring only those that support fractional hours and minutes),
+/// including ones affixed with `Z`, so such an operation is unnecessary.
+///
+/// ## Example
+///
+/// ```python
+/// from datetime import datetime
+///
+///
+/// date = "2025-01-01T00:00:00Z"
+///
+/// datetime.fromisoformat(date.replace("Z", "+00:00"))
+/// datetime.fromisoformat(date[:-1] + "-00")
+/// datetime.fromisoformat(date.strip("Z", "-0000"))
+/// datetime.fromisoformat(date.rstrip("Z", "-00:00"))
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// from datetime import datetime
+///
+///
+/// date = "2025-01-01T00:00:00Z"
+///
+/// datetime.fromisoformat(date)
+/// ```
+///
+/// ## Fix safety
+/// The fix is always marked as unsafe,
+/// as it might change the program's behaviour.
```

Comment:
Can we be more specific about how it could change program behavior? Ideally, the explanation would be detailed enough for users to decide if it's okay to mark the fix as safe in their project because they know this situation can't apply to their project.

Reply:
Oh I missed that, sorry. 

I'm still not entirely sure what the *barring only* part means. To me, it suggests that there are some cases where this operation isn't "safe". Do you have more context on what it means specifically?